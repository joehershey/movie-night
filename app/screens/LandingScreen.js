import React from "react";
import {
  ImageBackground,
  StyleSheet,
  View,
  Text,
  ScrollView,
  Image,
  TouchableWithoutFeedback,
  TextInput,
  Keyboard,
  Platform,
  SafeAreaView,
  Modal,
} from "react-native";

import { Card } from "react-native-elements";
import TopBar from "../components/TopBar";
import ButtonPanel from "../components/ButtonPanel";
import CreateGroupPopup from "../components/CreateGroupPopup";
import JoinGroupPopup from "../components/JoinGroupPopup";
import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";
import { useLinkProps } from "@react-navigation/native";
import { TEST_DATA } from "../assets/testData";

function LandingScreen(props) {
  const [isLoaded, setLoaded] = React.useState(false);
  const [showCreateGroup, toggleShowCreateGroup] = React.useState(false);
  const [showJoinGroup, toggleShowJoinGroup] = React.useState(false);

  const testGroups = [
    {
      group_name: "Spooky Sunday",
      group_id: 1,
      created_by: 1,
    },
    {
      group_name: "Mystery Monday",
      group_id: 2,
      created_by: 2,
    },
  ];

  const [groupsToRender, setGroups] = React.useState(testGroups);

  if (!isLoaded) {
    getGroupsAPI();
    setLoaded(!isLoaded);
  }

  function getGroupsAPI() {
    // fetch(props.url + "user/" + props.user_id + "/groups", {
    //     headers: {
    //       Accept: "application/json",
    //       "Content-Type": "application/json; charset=utf-8",
    //     },
    //     method: "GET",
    //   })
    //     .then((response) => response.json())
    //     .then((responseJson) => {
    //       () => setGroups(responseJson);
    //     })
    //     .catch((error) => {
    //       console.error(error);
    //     });
    () => setGroups(TEST_DATA.groups);
  }

  const groupsToRenderHTML = [];
  for (const [i, group] of groupsToRender.entries()) {
    //only non applied to properties

    groupsToRenderHTML.push(
      <Card>
        <Card.Title>{group.group_name}</Card.Title>
        <Card.Divider />
        <TouchableWithoutFeedback
          onPress={() => {
            props.setGroup(group.group_id);
            props.navigation.navigate("Schedule");
          }}
        >
          <View
            style={{
              margin: 10,
              padding: 5,
              backgroundColor: COLORS.primary,
              alignItems: "center",
              justifyContent: "center",
            }}
          >
            <Text
              style={{
                color: "white",
                fontSize: 20,
                alignContent: "center",
              }}
            >
              View Group
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </Card>
    );
  }

  return (
    <SafeAreaView style={STYLES.container}>
      <TopBar
        navigation={props.navigation}
        screen="Welcome"
        message="Log out"
      ></TopBar>

      <View style={styles.content}>
        <CreateGroupPopup
          showPopup={showCreateGroup}
          toggleShowPopup={toggleShowCreateGroup}
          url={props.url}
          user_id={props.user_id}
        ></CreateGroupPopup>
        <JoinGroupPopup
          showPopup={showJoinGroup}
          toggleShowPopup={toggleShowJoinGroup}
        ></JoinGroupPopup>
        <ScrollView
          style={{ width: "100%", marginTop: 20, marginBottom: 20 }}
          contentContainerStyle={{ alignItems: "center" }}
        >
          <View style={{ width: "90%" }}>{groupsToRenderHTML}</View>
        </ScrollView>
        {/* Button Panel start */}
        <ButtonPanel
          showCreateGroup={showCreateGroup}
          toggleShowCreateGroup={toggleShowCreateGroup}
          showJoinGroup={showJoinGroup}
          toggleShowJoinGroup={toggleShowJoinGroup}
        ></ButtonPanel>
        {/* Button Panel over */}
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  content: {
    //used with spacer to push content up the screen
    flex: 12,
    justifyContent: "center",
    alignItems: "center",
  },

  logo: {
    //to be added
    width: 275,
    height: 200,
    margin: 30,
  },
});

export default LandingScreen;
