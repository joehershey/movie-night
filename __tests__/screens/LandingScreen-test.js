import LandingScreen from "../../app/screens/LandingScreen";
import React from "react";
import renderer from "react-test-renderer";

describe("Test Landing Screen", () => {
  beforeEach(async () => {
    addListener = jest.fn();
    let props = {
      url: "https://sampleURL.com/",
      user_id: 1,
      navigation: {
        addListener
      }
    };
    global.tree = renderer.create(<LandingScreen {...props} />).toJSON();
  });

  it("Landing screen renders properly", async () => {
    console.error = jest.fn(); // fixme: mutes expected FetchError for testing purposes
    await expect(global.tree).toMatchSnapshot();
  });

});
