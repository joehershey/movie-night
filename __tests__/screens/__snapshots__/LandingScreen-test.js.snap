// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Landing Screen Landing screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        Log out
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingBottom": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginBottom": 0,
            "width": 100,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        User Profile
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 12,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <View>
        <Modal
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0.5)",
                "flex": 1,
                "justifyContent": "center",
                "marginTop": 22,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "borderColor": "#ccc",
                  "borderRadius": 20,
                  "elevation": 5,
                  "height": "50%",
                  "margin": 20,
                  "padding": 25,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 4,
                  "width": "90%",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 40,
                    "paddingBottom": 30,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 30,
                    }
                  }
                >
                  Create a Group
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Enter Group Name:
              </Text>
              <View
                style={
                  Object {
                    "height": 45,
                    "marginBottom": 20,
                    "width": 275,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Group Name"
                  placeholderTextColor="#D3D3D3"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": "auto",
                      "justifyContent": "center",
                      "margin": 4,
                      "padding": 10,
                      "width": "auto",
                    }
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Enter Your Alias for the Group (Optional):
              </Text>
              <View
                style={
                  Object {
                    "height": 45,
                    "marginBottom": 20,
                    "width": 275,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Alias"
                  placeholderTextColor="#D3D3D3"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": "auto",
                      "justifyContent": "center",
                      "margin": 4,
                      "padding": 10,
                      "width": "auto",
                    }
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#166fdb",
                        "height": 60,
                        "top": 10,
                        "width": "40%",
                      },
                      Object {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "justifyContent": "center",
                        "margin": 5,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignContent": "center",
                        "color": "white",
                        "fontSize": 20,
                      }
                    }
                  >
                    Submit
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1d1f1e",
                        "height": 60,
                        "top": 10,
                        "width": "40%",
                      },
                      Object {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "justifyContent": "center",
                        "margin": 5,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignContent": "center",
                        "color": "white",
                        "fontSize": 20,
                      }
                    }
                  >
                    Close
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </Modal>
        <Modal
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0.5)",
                "flex": 1,
                "justifyContent": "center",
                "marginTop": 22,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "borderColor": "#ccc",
                  "borderRadius": 20,
                  "elevation": 5,
                  "height": "50%",
                  "margin": 20,
                  "padding": 25,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 4,
                  "width": "90%",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 20,
                    "paddingBottom": 30,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 30,
                    }
                  }
                >
                  Successful Group Creation!
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Your group code is: . Use it to invite other members! You can also view it later in your Group Settings.
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1d1f1e",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Close
                </Text>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
    <View>
      <Modal
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.5)",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderColor": "#ccc",
                "borderRadius": 20,
                "elevation": 5,
                "height": "50%",
                "margin": 20,
                "padding": 25,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 40,
                  "paddingBottom": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                Join a Group
              </Text>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              Enter Group Code:
            </Text>
            <View
              style={
                Object {
                  "height": 45,
                  "marginBottom": 20,
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Group Code"
                placeholderTextColor="#D3D3D3"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "margin": 4,
                    "padding": 10,
                    "width": "auto",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              Enter Your Alias for the Group (Optional):
            </Text>
            <View
              style={
                Object {
                  "height": 45,
                  "marginBottom": 20,
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Alias"
                placeholderTextColor="#D3D3D3"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "margin": 4,
                    "padding": 10,
                    "width": "auto",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#166fdb",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Submit
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1d1f1e",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Close
                </Text>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "margin": 50,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontSize": 20,
                "fontStyle": "italic",
                "textAlign": "center",
                "top": 100,
              }
            }
          >
            You aren't currently a member of any groups, create one now or join an existing one with the group's access code!
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "90%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 30,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#166fdb",
              "height": 70,
              "top": 10,
              "width": "40%",
            },
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Create Group
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "height": 70,
              "top": 10,
              "width": "40%",
            },
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1d1f1e",
                "fontSize": 20,
              },
            ]
          }
        >
          Join Group
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
